Base/Python/slicer/util.py

```diff
@@ -2707,7 +2707,7 @@ def _messageDisplay(logLevel, text, testingReturnValue, mainWindowNeeded=False,
     if not windowTitle:
         windowTitle = slicer.app.applicationName + " " + logLevelString
     if slicer.app.testingEnabled():
-        logging.info("Testing mode is enabled: Returning %s and skipping message box [%s]." % (testingReturnValue, windowTitle))
+        logging.info(f"Testing mode is enabled: Returning {testingReturnValue} and skipping message box [{windowTitle}].")
```

Comment:
I would prefer not to introduce usage of f-strings in logging functions, because using f-strings results in performing expensive string manipulations even if the message does not end up being printed due to the current log level.

Has pyupgrade recommended this change?

Reply:
Yes `pyupgrade` changes it to .format() and then re-running changes it again to an f-string. The maintainer points to https://github.com/asottile/pyupgrade/issues/503#issuecomment-888267163 in that f-string usage here is not a performance drop.

Reply:
Thanks for the link @jamesobutler, it led to some very interesting discussions.  There seems to be a very strong disagreement between various Python developers on this topic and probably the correct answer depends on the use case. The insignificant performance drop was a measurement error, in the same issue further along the discussion, the actual performance drop was shown to be about 20%. So, this is a real problem. There are additional potential issues with exceptions during string conversion.

However, in our specific these cases here, there is no performance or exception-safety problems, so it should be fine to use f-strings.

My only concern is that using f-strings in logging would make this the default choice everywhere, even where performance matters. But maybe keeping the classic formatting in our log messages is not a very effective way of educating/reminding developers anyway, and if the cost is not to use pyupgrade at all or spend a lot of time with fighting with various lint tools, then it may just not worth it.

So, I'm OK with keeping the suggested changes.

Reply:
Looking at https://pylint.pycqa.org/en/latest/user_guide/messages/warning/logging-fstring-interpolation.html, the following is indicated:

> Another reasonable option is to use f-string. If you want to do that, you need to enable logging-format-interpolation and disable logging-fstring-interpolation.

Given the fact there is no significant performance impact [^1], enforcing f-string everywhere is reasonable.

That said, I think we should explicitly configure pylint with:
* `logging-fstring-interpolation` -> `disabled`
* `logging-format-interpolation` -> `enabled`

[^1]: https://github.com/Slicer/Slicer/pull/6376/files#r984071600